<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowTitle">
   <string>BeepConf Qt Chat</string>
  </property>
  <property name="minimumSize">
   <size><width>980</width><height>640</height></size>
  </property>

  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="rootLayout">
    <!-- Top bar: Prompts, GPT Version, Token Cost, Summary Pay -->
    <item>
     <layout class="QHBoxLayout" name="topBar">
      <item>
       <widget class="QPushButton" name="promptsButton">
        <property name="text"><string>Prompts</string></property>
       </widget>
      </item>
      <item>
       <spacer name="topStretch">
        <property name="orientation"><enum>Qt::Horizontal</enum></property>
        <property name="sizeHint"><size><width>40</width><height>20</height></size></property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="versionRow">
        <item><widget class="QLabel" name="gptVersionLabel"><property name="text"><string>GPT Version:</string></property></widget></item>
        <item>
         <widget class="QComboBox" name="model_combo">
          <item><property name="text"><string>gpt-4o-mini</string></property></item>
          <item><property name="text"><string>gpt-4o</string></property></item>
          <item><property name="text"><string>gpt-4</string></property></item>
          <item><property name="text"><string>gpt-3.5-turbo</string></property></item>
         </widget>
        </item>
      </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="tokenRow">
        <item><widget class="QLabel" name="tokenCostText"><property name="text"><string>Token Cost:</string></property></widget></item>
        <item><widget class="QLabel" name="tokenCostLabel"><property name="text"><string>$0.00</string></property></widget></item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="summaryRow">
        <item><widget class="QLabel" name="summaryPayText"><property name="text"><string>Summary Pay:</string></property></widget></item>
        <item><widget class="QLabel" name="summaryPayLabel"><property name="text"><string>$0.00</string></property></widget></item>
       </layout>
      </item>
     </layout>
    </item>

    <!-- Middle: left nav + chat area -->
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation"><enum>Qt::Horizontal</enum></property>

      <!-- Left navigation panel -->
      <widget class="QWidget" name="leftPanel">
       <layout class="QVBoxLayout" name="leftLayout">
        <item>
         <layout class="QHBoxLayout" name="navHeader">
          <item>
           <widget class="QToolButton" name="toggleNavButton">
            <property name="text"><string>â˜°</string></property>
            <property name="toolTip"><string>Hide and unhide nav navigation</string></property>
           </widget>
          </item>
          <item>
           <spacer name="navHeaderStretch">
            <property name="orientation"><enum>Qt::Horizontal</enum></property>
            <property name="sizeHint"><size><width>20</width><height>20</height></size></property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QListWidget" name="chatList">
          <property name="selectionMode"><enum>QAbstractItemView::SingleSelection</enum></property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="navFooter">
          <item>
           <widget class="QPushButton" name="settingsButton">
            <property name="text"><string>Settings</string></property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="customizeButton">
            <property name="text"><string>Customize</string></property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>

      <!-- Center chat panel -->
      <widget class="QWidget" name="centerPanel">
       <layout class="QVBoxLayout" name="centerLayout">
        <item>
         <widget class="QLabel" name="chatTitle">
          <property name="text"><string>Chat Title</string></property>
          <property name="styleSheet"><string notr="true">font-weight: 600; font-size: 18px;</string></property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="chat_history"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>

    <!-- Bottom input bar -->
    <item>
     <layout class="QHBoxLayout" name="inputBar">
      <item>
       <widget class="QPushButton" name="attach_btn">
        <property name="text"><string>Attach</string></property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="input_line">
        <property name="placeholderText"><string>Type your message...</string></property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="send_btn">
        <property name="text"><string>Send</string></property>
       </widget>
      </item>
     </layout>
    </item>

    <!-- Optional log area (collapsed by default; keep for debugging) -->
    <item>
     <widget class="QPlainTextEdit" name="log">
      <property name="readOnly"><bool>true</bool></property>
      <property name="maximumHeight"><number>110</number></property>
     </widget>
    </item>
   </layout>
  </widget>

  <widget class="QMenuBar" name="menubar"/>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
